### PR TITLE
AddTelegramLogic and func "rand"

### PR DESCRIPTION
**Pull request by HagBcemu**

### CODE REVIEW

File: MusicKeyStrokes/CommandsTelegram/ACommandTelegram.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        public abstract string CommandName { get; }
+
+        public abstract void Execute(Message message, TelegramBotClient client);
+        
+        public  bool Contains(string command)
+        {
+            return command.Contains(this.CommandName);
+        }
+    }
+}
```
Comment (Alexanderus32) [inline MusicKeyStrokes/CommandsTelegram/ACommandTelegram.cs]: Not needed

File: MusicKeyStrokes/CommandsTelegram/CommandTelegramRandom.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            this.audio = audio;
+        }
+
+        public override async void Execute(Message message, TelegramBotClient client)
+        {
+            audio.PLayRand();
+            await client.SendTextMessageAsync(message.Chat.Id,"Rand Ok");
+        }
+    }
+}
```
Comment (Alexanderus32) [inline MusicKeyStrokes/CommandsTelegram/CommandTelegramRandom.cs]: Not needed

File: MusicKeyStrokes/Program.cs [hunk 3]
```diff
@@ -22,6 +24,8 @@ static void Main()
             Application.EnableVisualStyles();
             Application.SetCompatibleTextRenderingDefault(false);
             CreateSimpleIngejector();
+            TelegramWatcher watcher = new TelegramWatcher();
+            watcher.WatchTelegram();
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Program.cs]: Need move to Form1

File: MusicKeyStrokes/Program.cs [hunk 4]
```diff
@@ -36,8 +40,8 @@ private static void CreateSimpleIngejector()
                     typeof(CommandAudioStop),
                     typeof(CommandAudioPlayRandMusic),
                     typeof(CommandAudioChangeLayout));
-            
-            
+            container.Collection.Register<ACommandTelegram>(
+                   typeof(CommandTelegramRandom));
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Program.cs]: not needed

File: MusicKeyStrokes/Telegram/TelegramClient.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Telegram.Bot;
+
+namespace MusicKeyStrokes
+{
+     static class TelegramClient
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Telegram/TelegramClient.cs]: delete static

File: MusicKeyStrokes/Telegram/TelegramClient.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        private static  TelegramBotClient  clientTelegram;
+
+         static public TelegramBotClient Get()
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Telegram/TelegramClient.cs:10]: public static and " "

File: MusicKeyStrokes/Telegram/TelegramWatcher.cs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+using System.Threading.Tasks;
+using Telegram.Bot;
+using Telegram.Bot.Args;
+using Telegram.Bot.Types.Enums;
+
+namespace MusicKeyStrokes.Telegram
+{
+    class TelegramWatcher
+    {
+        private static TelegramBotClient client;
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Telegram/TelegramWatcher.cs]: not needed static

File: MusicKeyStrokes/Telegram/TelegramWatcher.cs [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        {
+        //    if (!Status)
+        //    {
+                client.StartReceiving(new UpdateType[] { UpdateType.Message });
+                Thread.Sleep(100);
+            //    if (!Status)
+            //        client.StopReceiving();
+            //}
+        }
+
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Telegram/TelegramWatcher.cs]: not needed

File: MusicKeyStrokes/Commands/CommandAudioPauseBeforPlaying.cs [hunk 9]
```diff
@@ -12,16 +12,19 @@ public class CommandAudioPauseBeforPlaying : Command
     {
         public override string Name { get; }= Keys.D.ToString();
 
+        public override string Description => "Change property PauseBeforPlaying in Audio";
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Commands/CommandAudioPauseBeforPlaying.cs]: property change to status

File: MusicKeyStrokes/Commands/CommandGetCommand.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MusicKeyStrokes.Commands
+{
+    class CommandGetCommand : Command
+    {
+        public override string Name => "command";
+
+        public override string Description => "Gets command name and function";
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Commands/CommandGetCommand.cs]: Get list of commands

File: MusicKeyStrokes/Commands/CommandGetComputerName.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+        public override string Name => "computer";
+
+        public override string Description => "Get computer Name";
+
+        public override string Execute(string payload)
+        {
+            string domainname = Environment.UserDomainName;
+            string UserName = Environment.UserName;
+            string answer = $"{domainname} {UserName}";
+            return answer;
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Commands/CommandGetComputerName.cs:23]: Maybe we might more details of computer?

File: MusicKeyStrokes/Commands/CommandPlayMusic.cs [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+                int index = Int32.Parse(layoutCharMusic.ToString());
+                LayoutSound layoutSound = (LayoutSound)index;
+                audio.Play((Keys)KeyCharMusic, layoutSound);
+                return "Music Ok";
+            }
+            catch (Exception)
+            {
+
+                return "Music ops";
+            }
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Commands/CommandPlayMusic.cs]: if payload.length > 1 && int.tryparse(payload[1], int containsLayout)
if containsLayout > 0
 var key = payload[2];// this is key
else key =  payload[2];// this is key

File: MusicKeyStrokes/Commands/CommandsImageSayber.cs [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using MusicKeyStrokes.Interfaces;
+using MusicKeyStrokes.Models;
+using Newtonsoft.Json;
+
+namespace MusicKeyStrokes.Commands
+{
+    class CommandsImageSayber :Command
```
Comment (Alexanderus32) [inline MusicKeyStrokes/Commands/CommandsImageSayber.cs]: Saber

File: MusicKeyStrokes/Commands/CommandGetComputerName.cs (hunk 11 above)
Comment (HagBcemu) [reply MusicKeyStrokes/Commands/CommandGetComputerName.cs:23]: Make sense

**Pull request merged**